icu4j/main/classes/core/src/com/ibm/icu/impl/number/DecimalQuantity_AbstractBCD.java

```diff
@@ -216,6 +222,15 @@ public void adjustMagnitude(int delta) {
         }
     }
 
+    @Override
+    public int getOrigMagnitude() throws ArithmeticException {
+        DecimalQuantity_AbstractBCD copy = (DecimalQuantity_AbstractBCD) createCopy();
+        int magnitude = copy.getMagnitude();
+        int delta = -1 * magnitude;
+        copy.adjustMagnitude(delta);
+        return (-1 * copy.origDelta);
```

Comment:
If you're going to use `origDelta`, you need to be careful about when that value is reset to zero in `roundToMagnitude`.  You can probably remove the part where it resets to zero.  However, this is moot if you add a new `exponent` field instead.

Reply:
Done, as discussed.